dask/config.py

```diff
@@ -434,4 +436,40 @@ def update_defaults(new, config=config, defaults=defaults):
     update(config, new, priority='old')
 
 
+def expand_environment_variables(config):
+    ''' Utility to expand environment variables in a config dictionary
+
+    This function will recursively search through any nested dictionaries
+    and/or lists.
+
+    Parameters
+    ----------
+    config : dict, iterable, or str
+        Input object to search for environment variables
+
+    Returns
+    -------
+    config : same type as input
+
+    Examples
+    --------
+    >>> expand_environment_variables({'x': [1, 2, '$USER']})  # doctest: +SKIP
+    {'x': [1, 2, 'my-username']}
+    '''
+
+    orig_type = type(config)
+    if isinstance(config, Mapping):
+        new_config = orig_type()
```

Comment:
This operation isn't true for every mapping (`defaultdict` for example). I'm fine with it, but it might be better to just say `dict`. I'm not sure what the benefit of generic mappings is.

Reply:
this is a fair point. It would be true for an ordereddict though which was the main use case I had in mind.

Reply:
Are OrderedDicts common in configuration?  Is this something that you're running into @jhamman ?

I don't mind keeping this in, but I'm surprised that it came up.

(also, I resolved the graphviz thing)

Reply:
Nothing concrete. Happy to switch this line to a regular dict if that is the preference.

Reply:
Any preference @jcrist ?

Reply:
Slight preference to only support dicts, but could go either way.